### PR TITLE
Adding recursion

### PR DESCRIPTION
**Pull request by jkochanik**
zscp Recursion works consistently both ways now

### CODE REVIEW

File: zssh/zscp/root.go [hunk 1]
```diff
@@ -105,26 +122,93 @@ var (
...
+				logrus.Fatal(err, "error creating sftp client")
+			}
+			defer func() { _ = client.Close() }()
 
 			if isCopyToRemote {
-				zsshlib.SendFile(factory, localFilePath, remoteFilePath, svc)
+				if recursive {
+					baseDir := filepath.Base(localFilePath)
+					err := filepath.WalkDir(localFilePath, func(path string, info fs.DirEntry, err error) error {
+						remotePath := filepath.Join(remoteFilePath, baseDir, after(path, baseDir))
```
Comment (dovholuknf) [inline zssh/zscp/root.go]: what is `after` used for here? my initial reaction was "why is this here". it doesn't seem like it's needed?

Comment (jkochanik) [reply zssh/zscp/root.go]: returns the end of path after the base directory. Needed a slick way to make sure all the target directory path names didn't pile onto each other.

File: zssh/zscp/main.go [hunk 2]
```diff
@@ -1,5 +1,247 @@
...
+				remoteFilePath = args[0]
+				localFilePath = args[1]
+				isCopyToRemote = false
+
+			} else if strings.ContainsAny(args[1], ":") {
+				remoteFilePath = args[1]
+				localFilePath = args[0]
+				isCopyToRemote = true
+			} else {
+				logrus.Fatal("cannot determine remote file PATH use \":\" for remote path")
```
Comment (dovholuknf) [inline zssh/zscp/main.go]: for something like this you can use a string literal and not need to esacpe the quote character:

````
logrus.Fatal(`cannot determine remote file PATH use ":" for remote path`)
````

File: zssh/zscp/main.go [hunk 3]
```diff
@@ -1,5 +1,247 @@
...
+
+func init() {
+	rootCmd.Flags().StringVarP(&ZConfig, "ZConfig", "c", "", fmt.Sprintf("path to ziti config file. default: $HOME/.ziti/%s.json", serviceName))
+	rootCmd.Flags().StringVarP(&SshKeyPath, "SshKeyPath", "i", "", "path to ssh key. default: $HOME/.ssh/id_rsa")
+	rootCmd.Flags().BoolVarP(&debug, "debug", "d", false, "pass to enable additional debug information")
+	rootCmd.Flags().BoolVarP(&recursive, "recursive", "r", false, "pass to enable recursive file transfer")
+	rootCmd.Flags().StringVarP(&serviceName, "service", "s", ExpectedServiceAndExeName, fmt.Sprintf("service name. default: %s", ExpectedServiceAndExeName))
+}
+
+type ServiceConfig struct {
```
Comment (dovholuknf) [inline zssh/zscp/main.go]: doesn't seem used?

File: zssh/zssh/main.go [hunk 4]
```diff
@@ -1,6 +1,129 @@
...
+)
+
+func init() {
+	rootCmd.Flags().StringVarP(&serviceName, "service", "s", ExpectedServiceAndExeName, fmt.Sprintf("service name. default: %s", ExpectedServiceAndExeName))
+	rootCmd.Flags().StringVarP(&ZConfig, "ZConfig", "c", "", fmt.Sprintf("Path to ziti config file. default: $HOME/.ziti/%s.json", serviceName))
+	rootCmd.Flags().StringVarP(&SshKeyPath, "SshKeyPath", "i", "", "Path to ssh key. default: $HOME/.ssh/id_rsa")
+	rootCmd.Flags().BoolVarP(&debug, "debug", "d", false, "pass to enable additional debug information")
+}
+
+type ServiceConfig struct {
```
Comment (dovholuknf) [inline zssh/zssh/main.go]: doesn't seem used?

File: zssh/zscp/main.go [hunk 5]
```diff
@@ -1,5 +1,247 @@
...
+		return ""
+	}
+	adjustedPos := pos + len(a)
+	if adjustedPos >= len(value) {
+		return ""
+	}
+	return value[adjustedPos:len(value)]
+}
+
 func main() {
```
Comment (jkochanik) [inline zssh/zscp/main.go:239]: should I handle error  here
